Version 1.13: Update PX4-ROS2 and ROS(1) over ROS2 bridge instructions

beniaminopozzan opened the pull request:
- Instructions and links updated from `main` to `release/1.13` branches.
- Instructions for working on ROS (1) over the ROS2 bridge updated with message definitions synchronization step.

dagar commented:
I think we'd need to find a proper way to do this automatically. Do you want to limit this to just the ROS stuff and we could discuss the versioning approach with @hamishwillee separately?

beniaminopozzan commented:
> Do you want to limit this to just the ROS stuff?

Yes, of course, I've removed everything not ROS-related and adjusted the ROS pages to be point to the right versions.

hamishwillee commented:
@beniaminopozzan Thanks for this. Looks good. FYI for alignment of text in list it needs to be indented to the list item level.

I think we need to simplify this "for next time" by having a section in the build guide that covers getting a tag release. I'll add that now. So if you had to do this again we'd at least be able to link to a section on how to.

I'll tag you for review (and you can comment)

PS W.r.t. more generally keeping file paths up to date with a release this is not as easy as you might think. With gitbook legacy I could update links using a variable that I changed for the path, but vuepress does not support this.

I am (very slowly) looking at docsify for a possible alternative. That renders markdown on the fly, and would allow me to search for, and dynamically update the links based on the URL path. I think it will be possible, just hard for me to find time to do this as a "side job". FYI, my playpen here: https://hamishwillee.github.io/multidocsify_test/#/main/en/config/README

hamishwillee commented on en/ros/ros1_via_ros2.md:
```diff
@@ -72,6 +78,18 @@ As discussed in [ROS 2 User Guide > Sanity Check the Installation](../ros/ros2_c
 
 1. [Setup your PX4 Ubuntu Linux development environment](../dev_setup/dev_env_linux_ubuntu.md) - the default instructions get the latest version of PX4 source and install all the needed tools.
+
+   :::warning Note
```
```suggestion
   :::note
```

hamishwillee commented on en/ros/ros1_via_ros2.md:
````diff
@@ -72,6 +78,18 @@ As discussed in [ROS 2 User Guide > Sanity Check the Installation](../ros/ros2_c
+   The above instructions must be adjusted to get version v1.13 of PX4 source.
+   
+   ```sh
+   git clone -b <tag> https://github.com/PX4/PX4-Autopilot.git --recursive
````
@beniaminopozzan What you have as `tag` here is actually a branch - https://github.com/PX4/PX4-Autopilot/branches/all?query=release

Releases are tagged as well, but don't have the `release/` prefix: https://github.com/PX4/PX4-Autopilot/releases?q=release&expanded=true

What is it that we recommend people use?

hamishwillee replied:
Note, I THINK that the release branch will reflect latest-and-greatest, while the tag comes of that and reflects a specific point build. So we should suggest the branch "first".

beniaminopozzan replied:
I was a bit unsure about how to phrase that. My understanding is that the tags are the stable versions while the `release/` branches are where relevant commits are backported from `main` in preparation of an update of the releases, right?

So as this piece of doc is just for the release `1.13.*` I just listed all tags and branches which are compatible to it. If I had to choose, I'll say that `v1.13.2` it currently the recommended one.

beniaminopozzan commented:
> @beniaminopozzan Thanks for this. Looks good. FYI for alignment of text in list it needs to be indented to the list item level.

Thanks @hamishwillee  for the information, I'll be more careful next time. 

> I think we need to simplify this "for next time" by having a section in the build guide that covers getting a tag release. I'll add that now. So if you had to do this again we'd at least be able to link to a section on how to.

Yeah, a section about checking out specific releases is a good idea!

> PS W.r.t. more generally keeping file paths up to date with a release this is not as easy as you might think. With gitbook legacy I could update links using a variable that I changed for the path, but vuepress does not support this.

I'm sure of it, I was actually thinking that just search-replace all links and paths was quite naive.

> I am (very slowly) looking at docsify for a possible alternative. That renders markdown on the fly, and would allow me to search for, and dynamically update the links based on the URL path. I think it will be possible, just hard for me to find time to do this as a "side job". FYI, my playpen here: https://hamishwillee.github.io/multidocsify_test/#/main/en/config/README

You mean moving the whole documentation to a different platform?

hamishwillee commented:
> I'm sure of it, I was actually thinking that just search-replace all links and paths was quite naive.

It's fairly naïve, but not hopelessly so. That can help fix paths permanently to branch-specific locations, which is OK for links which can be tested, but not so much for text in code examples, where you might actually mean "mainline".
Upshot I probably should do more automation and updates of the old releases at release time, but its a maintenance cost, and there is only one of me.

> You mean moving the whole documentation to a different platform?

Yes. There are lots of reasons. No urgency though, which is why its a side project.
